File: test/lira_output/ts/notes_app/service.ts
// module notes_service
import { Note } from "./models";

import { NoteStore } from "./store";

export class NoteService {
  private store: NoteStore;
  constructor(store: NoteStore) {
    this.store = store;
  }
  public create(title: string): Note {
    let safeTitle: string = title;
    if ((title == "")) {
      safeTitle = "untitled";
    }
    let note: Note = new Note("1", safeTitle);
    this.store.save(note);
    return note;
  }
  public async load(id: string): Promise<Note> {
    let note: Note | null = this.store.get(id);
    if ((note == null)) {
      throw new Error("missing note");
    }
    return note;
  }
  public list(): Note[] {
    return this.store.list();
  }
  public firstTitle(): string {
    let notes: Note[] = this.store.list();
    let result: string = "";
    for (const note of notes) {
      if ((result == "")) {
        result = note.title;
      }
    }
    return result;
  }
}

File: test/lira_output/py/notes_app/service.py
# module notes_service
from "./models" import Note

from "./store" import NoteStore

class NoteService:
    store: NoteStore

    def __init__(self, store: NoteStore) -> None:
        self.store = store


    def create(self, title: str) -> Note:
        safeTitle = title
        if (title == ""):
            safeTitle = "untitled"
        note = Note("1", safeTitle)
        self.store.save(note)
        return note


    async def load(self, id: str) -> Note:
        note = self.store.get(id)
        if (note == None):
            raise Exception("missing note")
        return note


    def list(self) -> list[Note]:
        return self.store.list()


    def firstTitle(self) -> str:
        notes = self.store.list()
        result = ""
        for note in notes:
            if (result == ""):
                result = note.title
        return result
